apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Tests/V1/PgNumericTests.cs

```diff
@@ -329,56 +330,104 @@ public void FromDecimal(decimal input, string expectedValue)
+                "7.2345678901234567890123456789",
+                "8.234567890123456789012345678",
+                "0.8234567890123456789012345678",
+                "0.1234",
+                "12300000000"
+            )]
+            string text, LossOfPrecisionHandling handling, bool negate)
         {
-            var numeric = PgNumeric.Parse(input);
-            // decimal.Parse will silently lose precision if input has more significant digits than decimal can handle.
-            decimal expected = decimal.Parse(input, CultureInfo.InvariantCulture);
-            decimal actual = numeric.ToDecimal(LossOfPrecisionHandling.Truncate);
+            if (negate)
+            {
+                text = "-" + text;
+            }
+            PgNumeric numeric = PgNumeric.Parse(text);
+            decimal expected = decimal.Parse(text, CultureInfo.InvariantCulture);
+            decimal actual = numeric.ToDecimal(handling);
             Assert.Equal(expected, actual);
-            // Conversion via the explicit conversion should do the same thing.
-            decimal actual2 = (decimal)numeric;
-            Assert.Equal(expected, actual2);
+        }
+
+        [Theory, CombinatorialData]
+        public void ToDecimal_Overflow([CombinatorialValues(
+            "79228162514264337593543950336", // decimal.MaxValue + 1
+            "123456789012345678901234567890" // 30 integer places
+            )] string text, LossOfPrecisionHandling handling)
+        {
+            PerformTest(text, handling);
+            PerformTest("-" + text, handling);
+
+            static void PerformTest(string text, LossOfPrecisionHandling handling)
+            {
+                PgNumeric numeric = PgNumeric.Parse(text);
+                Assert.Throws<OverflowException>(() => numeric.ToDecimal(handling));
+            }
         }
 
         [Theory]
-        [InlineData("123456789012345678900.123456789", "123456789012345678900.12345679")]
         [InlineData("79228162514264337593543950335.0000000001", "79228162514264337593543950335")]
-        public void ToDecimal_LossOfPrecision_Truncate(string input, string expectedValue)
+        [InlineData("0.1234567891234567891234567899999", "0.1234567891234567891234567899")]
+        [InlineData("1.1234567891234567891234567899999", "1.1234567891234567891234567899")]
+        [InlineData("123456789012345678900.123456780", "123456789012345678900.12345678")]
+        [InlineData("123456789012345678900.123456785", "123456789012345678900.12345678")]
+        [InlineData("123456789012345678900.123456789", "123456789012345678900.12345678")]
+        [InlineData("123456789012345678900.123456770", "123456789012345678900.12345677")]
+        [InlineData("123456789012345678900.123456775", "123456789012345678900.12345677")]
+        [InlineData("123456789012345678900.123456779", "123456789012345678900.12345677")]
+        [InlineData("7.2345678901234567890123456789", "7.2345678901234567890123456789")]
```

Comment:
Possibly add a comment explaining the difference between this test case and the next one?